# HG changeset patch
# User Michael Godfrey <michaeldgodfrey@gmail.com>
# Date 1375575153 25200
#      Sat Aug 03 17:12:33 2013 -0700
# Node ID 0e34c475c956113cdc2f0f1c796d2746e660973c
# Parent  bb60b1bb47cff2ea2604df4de01a5d275a6d2106
plotimages.m: Add axis labels and titles to plots in Manual.

* doc/interpreter/plotimages.m: Add axis labels and titles to plots in Manual.

diff --git a/doc/interpreter/plotimages.m b/doc/interpreter/plotimages.m
--- a/doc/interpreter/plotimages.m
+++ b/doc/interpreter/plotimages.m
@@ -29,54 +29,75 @@ function plotimages (nm, typ)
     d_typ = ["-d", typ];
   endif
 
   if (strcmp(typ , "txt"))
     image_as_txt(nm);
   elseif (strcmp (nm, "plot"))
     x = -10:0.1:10;
     plot (x, sin (x));
+    xlabel ("x");
+    ylabel ("sin (x)");
+    title ("Simple 2-D Plot");
     print ([nm "." typ], d_typ);
   elseif (strcmp (nm, "hist"))
     rand ("state", 2);
     hist (randn (10000, 1), 30);
+    xlabel ("Value");
+    ylabel ("Count");
+    title ("Histogram of 10,000 normally distributed random numbers");
     print ([nm "." typ], d_typ);
   elseif (strcmp (nm, "errorbar"))
     rand ("state", 2);
     x = 0:0.1:10;
     y = sin (x);
     yl = 0.1 .* rand (size (x));
     yu = 0.1 .* rand (size (x));
     errorbar (x, sin (x), yl, yu);
     axis ([0, 10, -1.1, 1.1]);
+    xlabel ("x");
+    ylabel ("sin (x)");
+    title ("Errorbar plot of sin (x)");
     print ([nm "." typ], d_typ);
   elseif (strcmp (nm, "polar"))
     polar (0:0.1:10*pi, 0:0.1:10*pi);
+    title ("Example polar plot from 0 to 10*pi");
     print ([nm "." typ], d_typ);
   elseif (strcmp (nm, "mesh"))
     tx = ty = linspace (-8, 8, 41)';
     [xx, yy] = meshgrid (tx, ty);
     r = sqrt (xx .^ 2 + yy .^ 2) + eps;
     tz = sin (r) ./ r;
     mesh (tx, ty, tz);
+    xlabel ("tx");
+    ylabel ("ty");
+    zlabel ("tz");
+    title ("3-D Sombrero plot");
     print ([nm "." typ], d_typ);
   elseif (strcmp (nm, "plot3"))
     t = 0:0.1:10*pi;
     r = linspace (0, 1, numel (t));
     z = linspace (0, 1, numel (t));
     plot3 (r.*sin(t), r.*cos(t), z);
+    xlabel ("r.*sin (t)");
+    ylabel ("r.*cos (t)");
+    zlabel ("z");
+    title ("plot3 display of 3-D helix");
     print ([nm "." typ], d_typ);
   elseif (strcmp (nm, "extended"))
     x = 0:0.01:3;
-    plot(x,erf(x));
+    plot (x,erf(x));
     hold on;
-    plot(x,x,"r");
-    axis([0, 3, 0, 1]);
-    text(0.65, 0.6175, ['\leftarrow x = {2/\surd\pi {\fontsize{16}', ...
-      '\int_{\fontsize{8}0}^{\fontsize{8}x}} e^{-t^2} dt} = 0.6175']);
+    plot (x,x,"r");
+    axis ([0, 3, 0, 1]);
+    text (0.65, 0.6175, ['\leftarrow x = {2/\surd\pi {\fontsize{16}' \
+          '\int_{\fontsize{8}0}^{\fontsize{8}x}} e^{-t^2} dt} = 0.6175']);
+    xlabel ("x");
+    ylabel ("erf (x)");
+    title ("erf (x) with text annotation");
     print ([nm "." typ], d_typ);
   else
     error ("unrecognized plot requested");
   endif
   hide_output ();
 endfunction
 
 function set_print_size ()
